modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/NetCDFAuxiliaryStoreFactory.java

```diff
@@ -0,0 +1,236 @@
+
+import java.awt.RenderingHints.Key;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.JAXBException;
+
+import org.geotools.coverage.io.catalog.DataStoreConfiguration;
+import org.geotools.data.DataStore;
+import org.geotools.data.DataStoreFactorySpi;
+import org.geotools.data.FeatureReader;
+import org.geotools.data.FeatureWriter;
+import org.geotools.data.LockingManager;
+import org.geotools.data.Query;
+import org.geotools.data.ServiceInfo;
+import org.geotools.data.Transaction;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.imageio.netcdf.AncillaryFileManager;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.Name;
+import org.opengis.filter.Filter;
+import org.geotools.data.store.ContentDataStore;
+
+/**
+ * 
+ * Creates a vector store that publishes the index information of the NetCDF Store.
+ * This way the user can determine which combination of coordinates have data.
+ * 
+ * @author Niels CHarlier
+ *
+ */
+public class NetCDFAuxiliaryStoreFactory implements DataStoreFactorySpi {
+    
+    public static final Param FILE_PARAM = new Param("File", File.class, "NetCDF File Path", true,
+            null, Collections.emptyMap());
+    
+    public static final Param INDEX_PARAM = new Param("Index", String.class, "Index File Path", false,
+            null, Collections.emptyMap());    
+
+    public static final Param DS_PARAM = new Param("DataStore", String.class, "DataStore File Path", false,
+            null, Collections.emptyMap());
+    
+    public static final Param NS_PARAM = new Param("namespace", String.class, "Namespace", false,
+            null, Collections.emptyMap());
+
+    @Override
+    public String getDisplayName() {
+        return "NetCDF Auxiliary Store";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Published NetCDF Index Data";
+    }
+
+    @Override
+    public Param[] getParametersInfo() {
+        return new Param[] { FILE_PARAM, INDEX_PARAM, DS_PARAM, NS_PARAM };
+    }
+
+    @Override
+    public boolean canProcess(Map<String, Serializable> params) {
+        try {
+            File file = (File) FILE_PARAM.lookUp(params);
+            String indexPath = (String) INDEX_PARAM.lookUp(params);
+            String dsPath = (String) DS_PARAM.lookUp(params);
+            AncillaryFileManager ancilaryFileManager = new AncillaryFileManager(file, indexPath, dsPath);
+            DataStoreConfiguration datastoreConfig = ancilaryFileManager.getDatastoreConfiguration();
+            return datastoreConfig.getDatastoreSpi().canProcess(datastoreConfig.getParams());            
+        } catch (NoSuchAlgorithmException | JAXBException | IOException e) {
+            
```

Comment:
You should be able to move the ```return false``` to the ```catch``` block for clarity as all path then have a valid return.